src/retrieveDefaultValues.js

```diff
@@ -0,0 +1,9 @@
+exports.retrieveDefaultValues = (schema, instance) => Object.keys(schema).reduce((values, prop) => {
+  const defaultDecl = schema[prop].default;
+
+  if (defaultDecl) {
+    values[prop] = typeof defaultDecl === 'function' ? defaultDecl(instance) : defaultDecl;
```

Comment:
Maybe we should avoid calling this function always to then override the return value on `Object.assign`, thinking when 100 objects are instantiated at once it can be a performance killer.

Also, is there some way we can extract the function used in reduce or replace it with a normal `for`? It will create a new function for each new instance and it can make the performance worst too.